linera-service/src/exporter/runloops/task_manager.rs

```diff
@@ -0,0 +1,315 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    future::{Future, IntoFuture},
+    sync::Arc,
+    thread::JoinHandle,
+};
+
+use linera_client::config::{Destination, DestinationId, DestinationKind};
+use linera_rpc::{grpc::GrpcNodeProvider, NodeOptions};
+use linera_storage::Storage;
+use tokio::{
+    runtime::Handle,
+    task::{JoinError, JoinSet},
+};
+
+use crate::storage::ExporterStorage;
+
+/// This type manages tasks like spawning different exporters on the different
+/// threads, discarding the committees and joining every thread properly at the
+/// end for graceful shutdown of the process.
+pub(super) struct ThreadPoolState<F, S>
```

Comment:
I am not able to confirm this is working as intended so I am going to refactor it to shared multithreaded tokio Runtime.